core/src/main/scala/org/apache/spark/api/java/JavaRDD.scala

```diff
@@ -49,6 +50,20 @@ class JavaRDD[T](val rdd: RDD[T])(implicit val classTag: ClassTag[T])
    */
   def persist(newLevel: StorageLevel): JavaRDD[T] = wrapRDD(rdd.persist(newLevel))
 
+  /**
+   * Specify a ResourceProfile to use when calculating this RDD. This is only supported on
+   * certain cluster managers and currently requires dynamic allocation to be enabled.
+   * It will result in new executors with the resources specified being acquired to
+   * calculate the RDD.
+   */
+  def withResources(prof: ResourceProfile): JavaRDD[T] = wrapRDD(rdd.withResources(prof))
```

Comment:
Please use `@Private` annotation if this is meant to be private yet:
https://github.com/apache/spark/blob/9610efc252c94f93689d45e320df1c5815d97b25/common/tags/src/main/java/org/apache/spark/annotation/Private.java#L26-L28